### ISSUE TITLE
 12/5000 Confusion about inner and outer loops

### ISSUE THREAD

**Issue by zsszyx** [test-time-training/ttt-lm-pytorch#8]
The inner loop involves parameters k,v and the model weights but only updates the model weights. The outer loop involves parameters q and model weights， only qkv needs to be updated. How is it updated to kv?

**Comment by karan-dalal** [comment]
The "outer loop" updates QKV and the initialization of the "inner loop" model weights. We backprop through the updates to the inner model (gradients of gradients) to update the inner loop initialization.
